src/commonMain/kotlin/fr/acinq/lightning/channel/Channel.kt

```diff
@@ -550,8 +521,14 @@ sealed class ChannelStateWithCommitments : ChannelState() {
             }
         }
     }
+
+    companion object {
+
+    }
```

Comment:
That one is nasty. I can't remove the companion object because it is referenced indirectly by the external method in `Encryption.kt`:
```kotlin
fun ChannelStateWithCommitments.Companion.from(key: PrivateKey, encryptedChannelData: EncryptedChannelData): ChannelStateWithCommitments
```

I will add a comment. Alternatively I could have defined the external functions as normal functions in `ChannelStateWithCommitments` and in `EncryptedChannelData`, but I think it's better to put them at the same place.

Btw you make me realize that it should be `PersistedChannelState` here.

Reply:
Fixed in c5365f4.